pkg/anchor/witness/policy/policy.go

```diff
@@ -240,29 +240,31 @@ func (wp *WitnessPolicy) selectBatchAndSystemWitnesses(witnesses []*proof.Witnes
 
 	var selectedBatchWitnesses []*proof.Witness
 
-	var commonWitnessses []*proof.Witness
+	var commonWitnesses []*proof.Witness
 
 	if cfg.Operator == config.AND {
-		commonWitnessses = intersection(eligibleBatchWitnesses, eligibleSystemWitnesses)
+		commonWitnesses = intersection(eligibleBatchWitnesses, eligibleSystemWitnesses)
 	}
 
 	// it is possible to have 0 zero eligible batch witnesses
 	if len(eligibleBatchWitnesses) != 0 {
 		var err error
 
 		selectedBatchWitnesses, err = wp.selectMinWitnesses(eligibleBatchWitnesses, cfg.MinNumberBatch,
-			cfg.MinPercentBatch, totalBatchWitnesses, commonWitnessses...)
+			cfg.MinPercentBatch, totalBatchWitnesses, commonWitnesses...)
 		if err != nil {
-			return nil, nil, fmt.Errorf("select batch witnesses as per policy: %w", err)
+			return nil, nil, fmt.Errorf("select batch witnesses based on witnesses%s, eligible%s, common%s, total[%d], policy[%s]: %w", //nolint:lll
```

Comment:
Note that the lack of spaces between the words and `%s` may make this hard to read.

Reply:
Please disregard - I see now that they get printed with [] characters which makes them stand out